sentry_streams/src/gcs_writer.rs

```diff
@@ -55,35 +59,45 @@ impl GCSWriter {
 
         let client = client.default_headers(headers).build().unwrap();
 
-        GCSWriter { client, url }
+        GCSWriter { client, url, route }
     }
+}
 
-    pub fn write_to_gcs(
-        &self,
-        message: Message<RoutedValue>,
-    ) -> Result<Message<RoutedValue>, SubmitError<RoutedValue>> {
+impl TaskRunner<RoutedValue, RoutedValue, anyhow::Error> for GCSWriter {
+    fn get_task(&self, message: Message<RoutedValue>) -> RunTaskFunc<RoutedValue, anyhow::Error> {
         let client = self.client.clone();
         let url = self.url.clone();
-        let bytes = Python::with_gil(|py| pybytes_to_bytes(&message, py)).unwrap();
+        let route = message.payload().route.clone();
+        let actual_route = self.route.clone();
+
+        let bytes = traced_with_gil("writing to gcs", |py| pybytes_to_bytes(&message, py)).unwrap();
 
-        let res = client.post(&url).body(bytes).send();
+        Box::pin(async move {
+            if route != actual_route {
+                return Ok(message);
+            }
```

Comment:
I think this is quite inefficient when we need to forward the message to the next step. 
Let's say we have two routes: one send small messages somewhere and one sends batches to GCS.
This way we would use threads to send the small messages to the next step. 

I think we should customize the runTaskInThreads to avoid sending messages for other routes to the threadpool.

Though please file a task for this, no need to do it here